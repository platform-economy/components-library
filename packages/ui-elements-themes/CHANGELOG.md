# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.2](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.1...@relayr/ui-elements-themes@0.1.2) (2019-08-22)

**Note:** Version bump only for package @relayr/ui-elements-themes






## [0.1.1](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.0...@relayr/ui-elements-themes@0.1.1) (2019-08-21)

**Note:** Version bump only for package @relayr/ui-elements-themes






# 0.1.0 (2019-08-02)


### Features

* **ui-elements-basic:** new InputLabel component ([abb6076](https://github.com/relayr/ui-elements/commit/abb6076))
* **ui-elements-layout:** add Card component ([e8db0cf](https://github.com/relayr/ui-elements/commit/e8db0cf))
* **ui-elements-tables:** Table components ([#102](https://github.com/relayr/ui-elements/issues/102)) ([b1b8d6f](https://github.com/relayr/ui-elements/commit/b1b8d6f))
* **ui-elements-themes:** custom font linking system ([8e9f0f0](https://github.com/relayr/ui-elements/commit/8e9f0f0))
