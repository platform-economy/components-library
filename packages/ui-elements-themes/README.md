# ui-elements-themes

## Add to project

via npm:

```bash
npm install --save @relayr/ui-elements-themes
```

or Yarn:
```bash
yarn add @relayr/ui-elements-themes
```
