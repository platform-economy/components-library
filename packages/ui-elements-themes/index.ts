export * from './themes/light';
export * from './themes/dark';
export * from './themes/test';
export * from './themes/_common/paletteDark';
export * from './themes/_common/paletteLight';
export * from './themes/_common/methods';
export * from './themes/_common/fonts';
export * from './themes/_common/withTestTheme';
