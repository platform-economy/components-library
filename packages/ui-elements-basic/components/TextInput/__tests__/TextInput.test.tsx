import * as React from 'react';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import * as TestRenderer from 'react-test-renderer';

import 'jest-styled-components';

import { withTestTheme } from '@relayr/ui-elements-themes';

enzyme.configure({ adapter: new Adapter() });

import { TextInput } from '../index';

describe('ui-elements-basic', () => {
  describe('Components', () => {
    describe('TextInput', () => {
      const handleClick = jest.fn();
      it('should render with default state', () => {
        const tree = TestRenderer
          .create(withTestTheme(<TextInput />))
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
      it('renders conditional theme color based on given props', () => {
        const tree = TestRenderer
        .create(withTestTheme(
          <TextInput hollow={true}/>))
        .toJSON();
        expect(tree).toMatchSnapshot();
      });
      it('should render as a text input with all the optional properties', () => {
        const tree = TestRenderer
          .create(withTestTheme(
          <TextInput
            type="text"
            placeholder="test placeholder"
            value="test"
          />))
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('should correctly handle handleFocus function when reference exists', () => {
        const instance = enzyme.mount(withTestTheme(<TextInput/>));

        let error;

        try {
          instance.find('input').simulate('focus');
        } catch (err) {
          error = err;
        }

        expect(error).toBe(undefined);

      });

      it('should correctly handle handleBlur function when reference exists', () => {
        const instance = enzyme.mount(withTestTheme(<TextInput/>));

        let error;

        try {
          instance.find('input').simulate('blur');
        } catch (err) {
          error = err;
        }

        expect(error).toBe(undefined);

      });

      it('should correctly handle handleClick function when reference exists', () => {
        const instance = enzyme.mount(withTestTheme(<TextInput onClick={handleClick}/>));

        let error;

        try {
          instance.find('input').simulate('click');
        } catch (err) {
          error = err;
        }

        expect(error).toBe(undefined);
        expect(handleClick).toHaveBeenCalledTimes(1);
      });
      it('should correctly handle clear click', () => {
        const wrapperComponent = (props: { children: React.ReactElement }) =>
        withTestTheme(props.children);
        const tree = enzyme.mount(
          <TextInput clearIcon={'text'}/>,
          {
            wrappingComponent: wrapperComponent,
          },
        );
        const input = tree.find('input');
      â€‹
        input.getDOMNode<HTMLInputElement>().value = 'foo';
        expect(input.getDOMNode<HTMLInputElement>().value).toBe('foo');
        tree.find('i').simulate('click');
        expect(input.getDOMNode<HTMLInputElement>().value).toBe('');
      });
    });
  });
});
