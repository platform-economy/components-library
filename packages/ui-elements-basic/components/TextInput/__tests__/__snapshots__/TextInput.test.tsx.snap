// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-elements-basic Components TextInput renders conditional theme color based on given props 1`] = `
.c0 {
  font-size: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  height: 35px;
  box-shadow: none;
  background-color: transparent;
  color: #text;
  cursor: text;
  border: none;
  margin: 0;
  padding: 10px;
  padding-right: 10px;
  box-sizing: border-box;
  border-bottom: 2px solid #text_inverted;
}

.c0:focus {
  border-bottom: 2px solid #active;
  outline: transparent;
}

.c0:invalid {
  border-bottom: 2px solid #negative;
  outline: transparent;
}

.c0 svg,
.c0 i {
  box-sizing: border-box;
  height: 100%;
  margin: 0 10px;
}

.c0 + i {
  font-size: 1em;
  position: absolute;
  top: 9px;
  right: 10px;
  -webkit-transition: all .3s ease;
  transition: all .3s ease;
  color: #text_inverted;
  cursor: pointer;
  pointer-events: auto;
}

.c0:placeholder-shown + i {
  cursor: text;
  pointer-events: none;
  opacity: 0;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.c0 + i:hover {
  color: #text;
}

@media screen and (min-width:414px) {
  .c0::-webkit-input-placeholder {
    font-size: .875em;
  }

  .c0::-moz-placeholder {
    font-size: .875em;
  }

  .c0:-ms-input-placeholder {
    font-size: .875em;
  }

  .c0::-webkit-input-placeholder {
    font-size: .875em;
  }

  .c0::-moz-placeholder {
    font-size: .875em;
  }

  .c0:-ms-input-placeholder {
    font-size: .875em;
  }

  .c0::placeholder {
    font-size: .875em;
  }
}

<input
  className="c0"
/>
`;

exports[`ui-elements-basic Components TextInput should render as a text input with all the optional properties 1`] = `
.c0 {
  font-size: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  height: 35px;
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.23);
  background-color: #foreground;
  color: #text;
  cursor: text;
  border: none;
  margin: 0;
  padding: 10px;
  padding-right: 10px;
  box-sizing: border-box;
  border-bottom: 2px solid transparent;
}

.c0:focus {
  border-bottom: 2px solid #active;
  outline: transparent;
}

.c0:invalid {
  border-bottom: 2px solid #negative;
  outline: transparent;
}

.c0 svg,
.c0 i {
  box-sizing: border-box;
  height: 100%;
  margin: 0 10px;
}

.c0 + i {
  font-size: 1em;
  position: absolute;
  top: 9px;
  right: 10px;
  -webkit-transition: all .3s ease;
  transition: all .3s ease;
  color: #text_inverted;
  cursor: pointer;
  pointer-events: auto;
}

.c0:placeholder-shown + i {
  cursor: text;
  pointer-events: none;
  opacity: 0;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.c0 + i:hover {
  color: #text;
}

@media screen and (min-width:414px) {
  .c0::-webkit-input-placeholder {
    font-size: .875em;
  }

  .c0::-moz-placeholder {
    font-size: .875em;
  }

  .c0:-ms-input-placeholder {
    font-size: .875em;
  }

  .c0::-webkit-input-placeholder {
    font-size: .875em;
  }

  .c0::-moz-placeholder {
    font-size: .875em;
  }

  .c0:-ms-input-placeholder {
    font-size: .875em;
  }

  .c0::placeholder {
    font-size: .875em;
  }
}

<input
  className="c0"
  placeholder="test placeholder"
  type="text"
  value="test"
/>
`;

exports[`ui-elements-basic Components TextInput should render with default state 1`] = `
.c0 {
  font-size: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  height: 35px;
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.23);
  background-color: #foreground;
  color: #text;
  cursor: text;
  border: none;
  margin: 0;
  padding: 10px;
  padding-right: 10px;
  box-sizing: border-box;
  border-bottom: 2px solid transparent;
}

.c0:focus {
  border-bottom: 2px solid #active;
  outline: transparent;
}

.c0:invalid {
  border-bottom: 2px solid #negative;
  outline: transparent;
}

.c0 svg,
.c0 i {
  box-sizing: border-box;
  height: 100%;
  margin: 0 10px;
}

.c0 + i {
  font-size: 1em;
  position: absolute;
  top: 9px;
  right: 10px;
  -webkit-transition: all .3s ease;
  transition: all .3s ease;
  color: #text_inverted;
  cursor: pointer;
  pointer-events: auto;
}

.c0:placeholder-shown + i {
  cursor: text;
  pointer-events: none;
  opacity: 0;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.c0 + i:hover {
  color: #text;
}

@media screen and (min-width:414px) {
  .c0::-webkit-input-placeholder {
    font-size: .875em;
  }

  .c0::-moz-placeholder {
    font-size: .875em;
  }

  .c0:-ms-input-placeholder {
    font-size: .875em;
  }

  .c0::-webkit-input-placeholder {
    font-size: .875em;
  }

  .c0::-moz-placeholder {
    font-size: .875em;
  }

  .c0:-ms-input-placeholder {
    font-size: .875em;
  }

  .c0::placeholder {
    font-size: .875em;
  }
}

<input
  className="c0"
/>
`;
